src/clj/y_video_back/routes/service_handlers/handlers/file_handlers.clj

```diff
@@ -8,34 +8,49 @@
    [y-video-back.model-specs :as sp]
    [y-video-back.routes.service-handlers.utils.utils :as utils]
    [reitit.ring.middleware.multipart :as multipart]
+   [clojure.data.json :as json]
    [clojure.java.io :as io]
-   [clojure.java.shell :as shell]))
+   [clojure.java.shell :as shell]
+   [ffclj.core :as ffc]))
 
-(def file-create
-  {:summary "Creates a new file. MUST INCLUDE FILE AS UPLOAD."
-   :permission-level "lab-assistant"
-   :parameters {:header {:session-id uuid?}
-                :multipart {:file multipart/temp-file-part
-                            :resource-id uuid?
-                            :file-version string?
-                            :metadata string?}}
-   :responses {200 {:body {:message string?
-                           :id string?}}
-               500 {:body {:message string?}}}
-   :handler (fn [{{{:keys [file resource-id file-version metadata]} :multipart} :parameters}]
-              (let [file-name (utils/get-filename (:filename file))]
+(defn compute-aspect-ratio
+  "given width and height, calculate the aspect ratio of them; return a vector or an 'x:y' string version"
+  [w h &[string?]]
+  (let [calc-gcd (fn [a b]
+              (if (zero? b)
+                a
+                (recur b (mod a b))))
+        [x y :as xy] (map (comp #(Integer/parseInt %) str) [w h])
+        gcd (calc-gcd x y)
+        [rx ry :as xy] (map #(/ % gcd) xy)]
+    (if string?
+      (str rx ":" ry)
+      xy)))
```

Comment:
I'm going to log this away as the perfect example of how unreadable Clojure is. XD